src/scripts/main.js

```diff
@@ -1,3 +1,329 @@
+const cells = document.querySelectorAll('.field-cell');
+
+const cellsData = [];
+let isGameOver = false;
+let score = 0;
+
+// Create an array for each cell
+for (let i = 0; i < 16; i++) {
+  cellsData.push({ value: 0 });
+}
+
+// Add a new number to the playing field
+function addNewNumber() {
+  // Get all free cells
+  const emptyCells = cellsData.filter((cell) => cell.value === 0);
+
+  if (emptyCells.length > 0) {
+    // Select a random free cell
+    const randomIndex = Math.floor(Math.random() * emptyCells.length);
+    const randomCell = emptyCells[randomIndex];
+    // const randomCell1 = emptyCells[randomIndex + 1];
+    // const randomCell2 = emptyCells[randomIndex + 2];
+
+    // randomCell1.value = 1024;
+    // randomCell2.value = 1024;
+    // Set value to 2 or 4
+    randomCell.value = Math.random() < 0.9 ? 2 : 4;
+
+    scoreDisplay.textContent = score;
+    // Update cell data on the page
+
+    updateCells();
+  }
+}
+
+function updateCells() {
+  cells.forEach((cell, index) => {
+    const cellValue = cellsData[index].value;
+
+    if (cellValue !== 0) {
+      cell.textContent = cellValue;
+
+      cell.classList.remove(
+        'field-cell--2',
+        'field-cell--4',
+        'field-cell--8',
+        'field-cell--16',
+        'field-cell--32',
+        'field-cell--64',
+        'field-cell--128',
+        'field-cell--256',
+        'field-cell--512',
+        'field-cell--1024',
+        'field-cell--2048'
+      );
+      cell.classList.add(`field-cell--${cellValue}`);
+    } else {
+      cell.textContent = '';
+
+      cell.classList.remove(
+        'field-cell--2',
+        'field-cell--4',
+        'field-cell--8',
+        'field-cell--16',
+        'field-cell--32',
+        'field-cell--64',
+        'field-cell--128',
+        'field-cell--256',
+        'field-cell--512',
+        'field-cell--1024',
+        'field-cell--2048'
+      );
+    }
+  });
+}
+
+startButton.addEventListener('click', () => {
+  resetGame();
+  addNewNumber();
+  startButton.textContent = 'Restart';
+  startButton.classList.add('restart');
+  startMessage.classList.add('hidden');
+});
+
+function resetGame() {
+  score = 0;
+
+  // Reset all cell values to 0
+  cellsData.forEach((cell) => {
+    cell.value = 0;
+  });
+
+  isGameOver = false;
+
+  // Clear score display and messages
+  scoreDisplay.textContent = '0';
+  loseMessage.classList.remove('show');
+  winMessage.classList.remove('show');
+
+  // Remove cell classes
+  cells.forEach((cell) => {
+    cell.textContent = '';
+
+    cell.classList.remove(
+      'field-cell--2',
+      'field-cell--4',
+      'field-cell--8',
+      'field-cell--16',
+      'field-cell--32',
+      'field-cell--64',
+      'field-cell--128',
+      'field-cell--256',
+      'field-cell--512',
+      'field-cell--1024',
+      'field-cell--2048'
+    );
+  });
+
+  addNewNumber();
+}
+
+function checkWin() {
+  for (let row = 0; row < cellsData.length; row++) {
```

Comment:
Use functional programming constructs like `forEach` or `reduce` instead of `for` loops for iterating over arrays. You can use `some` to check if there's a cell with a value of 2048. Example: `return cellsData.some(cell => cell.value === 2048);`